test/framework/src/main/java/org/opensearch/tracing/MockSpan.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.opensearch.common.Randomness;
+import org.opensearch.telemetry.tracing.AbstractSpan;
+import org.opensearch.telemetry.tracing.Span;
+
+/**
+ * MockSpan for testing and strict check validations. Not to be used for production cases.
+ */
+public class MockSpan extends AbstractSpan {
```

Comment:
I think this should go to test library, like we have for client, :libs:opensearch-telemetry-test

Reply:
Thanks @reta for reviewing. Since currently all the mock plugins are in test framework so kept in test subproject for now but we can discuss about the best place for this. AFAIK, libs shouldn't have concrete implementations. What do you suggest? Also, pls go through rest of the changes as well.

Reply:
You literally mock the while library (opensearch-telemetry), I think there is no need to inflate the test framework with that, you may probably keep the plugin but everything else should be out I think.

Also, why do we need this plugin to be present by default? The test framework is used by mostly any other OpenSearch project (fe `opensearch-java`) which do not deal with tracing yet.

Reply:
The only reason is, if some project starts using the opensearch-telemetry then we would want to make sure that all the span strict checks are validated. But yes, we can extract out the mock plugin implementation to the test library and add it as a dependency to test project. Your thoughts?

Reply:
> But yes, we can extract out the mock plugin implementation to the test library and add it as a dependency to the test project.

:+1: 

> Overall what's your opinion on enforcing the strict checks like -
> 1. All the spans should be ended.
> 2. Parents shouldn't be closed before their child.

Certainly :+1:

Reply:
Implementing the mock plugin in libs is tricky because it requires dependency of server and opensearch-telemetry. So for now, I have abstracted it out from `test:framework` to a `test:telemetry`. We can decide the best place may be later on.

Reply:
> Implementing the mock plugin in libs is tricky because it requires dependency of server

Correct, I think the the place for plugin would be `test:framework` at the moment

Reply:
I have extracted it out to a  `test:telemetry`. I think it should be good for now.